pkg/apis/pipeline/v1beta1/taskrun_conversion.go

```diff
@@ -193,6 +199,150 @@ func (trso *TaskRunSidecarOverride) convertFrom(ctx context.Context, source v1.T
 	trso.Resources = source.ComputeResources
 }
 
+// ConvertTo implements apis.Convertible
+func (trs *TaskRunStatus) ConvertTo(ctx context.Context, sink *v1.TaskRunStatus) error {
+	// TODO: handle duckv1beta1 to v1
+	sink.PodName = trs.PodName
+	sink.StartTime = trs.StartTime
+	sink.CompletionTime = trs.CompletionTime
+	sink.Steps = nil
+	for _, ss := range trs.Steps {
+		new := v1.StepState{}
+		ss.convertTo(ctx, &new)
+		sink.Steps = append(sink.Steps, new)
+	}
+	// CloudEvents are dropped
```

Comment:
Shall we just drop `CloudEvents` and `ResourcesResult`?

Reply:
Confirmed at WG and https://github.com/tektoncd/pipeline/issues/5785 that this shall not be just dropped. Next step to decide if also serializing `CloudEvents` and `ResourcesResult` could be the solution here.